spx-gui/src/components/editor/sprite/animation/AnimationPlayer.vue

```diff
@@ -59,7 +59,10 @@ const preloadFrames = async (costumeFiles: File[]) => {
       ...urls.map(async (url) => {
         const img = new Image()
         img.src = url
-        await img.decode()
+        await img.decode().catch((e) => {
+          // Sometimes `decode` fails, while the image is still able to be displayed
```

Comment:
这里的 `decode` 操作是起到预解码的作用吗？如果预解码失败了，渲染时浏览器会再以不同的方式尝试解码？

Reply:
> 这里的 `decode` 操作是起到预解码的作用吗？

是的

> 如果预解码失败了，渲染时浏览器会再以不同的方式尝试解码？

如果是格式导致的解码失败，没理由渲染的时候跟 decode 的时候解码逻辑不同..

我猜测是图片本身过大，导致对应的 bitmap 内存占用大，浏览器内存管理的相关策略 reject 了 decode，只是以 `EncodingError` 的名义抛出来了。[spec](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-decode) 里有提到

> User agents should ensure that the decoded media data stays readily available until at least the end of the next successful [update the rendering](https://html.spec.whatwg.org/multipage/webappapis.html#update-the-rendering) step in the [event loop](https://html.spec.whatwg.org/multipage/webappapis.html#event-loop). This is an important part of the API contract, and should not be broken if at all possible. (Typically, this would only be violated in low-memory situations that require evicting decoded image data, or when the image is too large to keep in decoded form for this period of time.)

不过还只是猜测

Reply:
> 我猜测是图片本身过大，导致对应的 bitmap 内存占用大

这几张图片都是大概 5000x6000 大小的

Reply:
明白了